### ISSUE TITLE
Build using tensorrt for c# failing

### ISSUE THREAD

**Issue by MaxHuerlimann** [microsoft/onnxruntime#5198]
**Describe the bug**
I am trying to build onnxruntime with tensorrt support for C#, as there is no official nuget package with tensorrt support. I am doing so within a docker container, following mostly [this dockerfile](https://github.com/microsoft/onnxruntime/blob/master/dockerfiles/Dockerfile.tensorrt) but adding the necessary dependencies for C# (dotnet, mono, nuget.exe). I am using the following command for the build:
`bin/sh ./build.sh --cuda_home /usr/local/cuda --skip_tests --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_tensorrt  --build_nuget --tensorrt_home /workspace/tensorrt --config Release --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) 
`
The build finishes but while installing the nuget components the following error pops up (note that this is from a second run of the build process):
```-- Up-to-date: ./nuget-staging/usr/local/include/onnxruntime/core/framework/func_api.h
-- Up-to-date: ./nuget-staging/usr/local/include/onnxruntime/core/framework/op_kernel_info.h
-- Up-to-date: ./nuget-staging/usr/local/include/onnxruntime/core/providers/cpu
-- Up-to-date: ./nuget-staging/usr/local/include/onnxruntime/core/providers/cpu/cpu_provider_factory.h
-- Up-to-date: ./nuget-staging/usr/local/include/onnxruntime/core/providers/cuda
-- Up-to-date: ./nuget-staging/usr/local/include/onnxruntime/core/providers/cuda/cuda_provider_factory.h
CMake Error at cmake_install.cmake:61 (file):
  file INSTALL cannot find
  "/code/onnxruntime/cmake/../include/onnxruntime/core/providers/shared".
```
The respective folder does not exist in the repository. I am not too familiar with cmake and therefore not sure how to handle this.

**Urgency**
None

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04 (within docker)
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: 1.4.0
- Python version: 3.6
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source): 7.5.0
- CUDA/cuDNN version: 11.0.3, 8.0.2 
- GPU model and memory: GeForce RTX 2060 maxQ

**To Reproduce**
Build following dockerfile
```# --------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.
# --------------------------------------------------------------
# Dockerfile to run ONNXRuntime with TensorRT integration

# nVidia TensorRT Base Image
FROM nvcr.io/nvidia/tensorrt:20.07.1-py3

ARG ONNXRUNTIME_REPO=https://github.com/Microsoft/onnxruntime
ARG ONNXRUNTIME_BRANCH=master

RUN apt-get update &&\
    apt-get install -y sudo git bash unattended-upgrades
RUN unattended-upgrade

WORKDIR /code
ENV PATH /usr/local/nvidia/bin:/usr/local/cuda/bin:/code/cmake-3.14.3-Linux-x86_64/bin:/opt/miniconda/bin:${PATH}
ENV LD_LIBRARY_PATH /opt/miniconda/lib:$LD_LIBRARY_PATH

# Prepare onnxruntime repository & build onnxruntime with TensorRT
RUN git clone --single-branch --branch ${ONNXRUNTIME_BRANCH} --recursive ${ONNXRUNTIME_REPO} onnxruntime &&\
    /bin/sh onnxruntime/dockerfiles/scripts/install_common_deps.sh &&\
    cp onnxruntime/docs/Privacy.md /code/Privacy.md &&\
    cp onnxruntime/dockerfiles/LICENSE-IMAGE.txt /code/LICENSE-IMAGE.txt &&\
    cp onnxruntime/ThirdPartyNotices.txt /code/ThirdPartyNotices.txt 

# Install mono for builiding nuget
ENV TZ=Europe/Rome
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
RUN sudo apt-get install -y mono-devel
RUN sudo apt-get install -y mono-complete
RUN sudo curl -o /usr/local/bin/nuget.exe https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
RUN echo 'alias nuget="mono /usr/local/bin/nuget.exe"' >> /home/.bash_aliases
RUN source /home/.bash_aliases

# Install dotnet
RUN wget https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
RUN sudo dpkg -i packages-microsoft-prod.deb
RUN sudo apt-get update; \
  sudo apt-get install -y apt-transport-https && \
  sudo apt-get update
RUN sudo apt-get install -y dotnet-sdk-3.1
RUN sudo apt-get install -y aspnetcore-runtime-3.1

RUN cd onnxruntime &&\
    /bin/sh ./build.sh --cuda_home /usr/local/cuda --skip_tests --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_tensorrt --build_nuget --tensorrt_home /workspace/tensorrt --config Release --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) &&\
    # cp -r build/ /home &&\
    cd .. &&\
    rm -rf cmake-3.14.3-Linux-x86_64
    # rm -rf onnxruntime cmake-3.14.3-Linux-x86_64

RUN sudo apt-get install -y libgomp1
RUN sudo apt-get install -y libgdiplus
```

**Expected behavior**
The build should succeed

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Comment by codemzs** [comment]
@snnn might be able to help with build issues.

**Comment by snnn** [comment]
It should have been fixed by #5219
